### ISSUE TITLE
Avoid using non-inclusive language

### ISSUE THREAD

**Issue by slyon** [systemd/systemd#26257]
### Component

systemd-networkd

### Is your feature request related to a problem? Please describe

Systemd-networkd is using non-inclusive language, such as the term "slave", e.g. in configuration options such as `AllSlavesActive=`, `PacketPerSlave=` or `PrimarySlave=`.

### Describe the solution you'd like

Maybe the term "slave" should be replaced by the term "member" in a networking context (or at least provided as an alias), similar to how Open vSwitch (https://github.com/openvswitch/ovs/commit/91fc374a9c5a2a4d9520aaa3588a7a18338a476e) or Netplan (https://github.com/canonical/netplan/pull/303) have done.

This might also apply to more terms, even across all of systemd and should be checked against https://inclusivenaming.org/ using the [woke](https://docs.getwoke.tech/) linter.

### Describe alternatives you've considered

Other inclusive alternatives for the term "slave" might be "parent/child", "secondary" or "replica", according to https://github.com/canonical/Inclusive-naming/blob/main/config.yml

### The systemd version you checked that didn't have the feature you are asking for

v252

**Comment by bluca** [comment]
if networkmanager/netplan/ovs are all picking new names for the same functionality, and they all use the same ones, then I think it's fine to move as well - we can make the kernel the odd one out. In the end, users see userspace applications and rarely interact with the kernel directly. Did the other project switch to a common naming scheme?

**Comment by poettering** [comment]
Sure, if everyone but the kernel people agree, I guess we can switch this over too.

What about iproute2?

**Comment by bluca** [comment]
I don't think it has been updated yet - @slyon is that on your radar too?

**Comment by slyon** [comment]
I reached out to NetworkManager (https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/1205) and they want to do the switch as well, but are considering using the term `ports` instead of `members` (as used in OVS and Netplan). I did not reach out to the iproute project.

**Comment by keszybz** [comment]
I'd vote for renaming (with backwards-compat provisions in place). It would be great if we could use the same name as other configuration tools. I expect that the kernel will move very slowly, but we don't really need to wait for it.  Once we have an agreement on the new names, the actual patches to introduce them are simple.

**Comment by yuwata** [comment]
So, the problem here is only "slave"? Are there any other words we should replace?

**Comment by keszybz** [comment]
"dummy", seriously? I'd be happy to get rid of "slave"/"master", but I'm not convinced about stuff like "dummy".

**Comment by slyon** [comment]
Well, that's what "woke" / https://inclusivenaming.org/ suggests. But no need to force that into systemd. Avoiding the term "slave" would be a big step already IMO

**Comment by premyslc** [comment]
omg

**Comment by g0tar** [comment]
networkd is using technical language, if anyone projects the well-known and recognized terms to layman human dictionary then the problem needs to be resolved by his therapist with the strong advise not to read something beyond his comprehension, not by forcing entire world to adapt to his phobias.
.
As for the scientific approach: https://en.wiktionary.org/wiki/slave#Etymology - if anyone should be offended by "slave" it is the **Slavs** - Zbyszek, you are Slav (and **I am** too), do you feel offended? I don't.

But I do feel offended when the BLM-like people are posing black to be the only slaves in the history of the world, pretending that any other harmful actions from the past were less important. This way they are actually stealing the sense of justice by pretending they deserve a special treatment, while in fact their legacy is not uncommon. On the contrary, slavery was ubiquitous.

And the term itself is perfect for what it describes - enslaving device.

What makes "black"(list) so special to be offending? Is red traffic light non-inclusive for native Americans? Then the green and blue might be offending to Martians. Well, there is actually `net.ipv4.conf.*.log_martians` which offends possible ancient (or future) civilization of the Red Planet.

This PC madness must simply stop...

**Comment by yuwata** [comment]
I agree with most of the above @g0tar's comment, and I feel I cannot wholeheartedly support the recent movement.

However, at the same time, the most important thing here is neither history nor linguistics, but if we can reword master/slave (and/or any other words) without losing its context. And if so, then there is nothing to prevent us from renaming the words and no reason to keep the old terminology. In that sense, "slave" -> "port" conversion should be OK. And, I've not checked other tools yet by myself, but if I believe the above comments, some tools are already moving to use "ports". So, at least for the words master/slave, I think we can change them.

**Comment by bluca** [comment]
> networkd is using technical language, if anyone projects the well-known and recognized terms

These sort of comments are not welcome here. This is the first and only warning, next time you will be banned from the org. No ifs, no buts, just out. This is not up for discussion, and you can leave at any time if you don't like it. Are we clear?

**Comment by g0tar** [comment]
The problem with "port" (or "member" or anything that is inherently "equal" to some "master") is that enslaved network devices **are** enslaved and **are not** autonomous, they are **ruled** by whoever claimed them. They might be equal between themselves, but are servant to their masters - just like the "slaves". That is why this word was picked in the first place (it that wasn't clear enough for anybody, Captain Obvious to the rescue!).

The port can be managed independently. **Port** is the **physical (give or take) entity**.
Two+ ports bonded (like LACP) are not constituting new port. Slave is the **logical function** of the **port**.

Changing "slave" to "port" would be like changing "woman" to "man" (equal to "human").

I'm not fluent enough in english to find a better word, but the "minion" seems to fit better;)


If we're going to pretend that any word describing lack of freedom (like captivity or imprisonment) might be unpleasant to somebody (not because of his **own** bad memories, rather some social psychosis - I doubt any former real slave manages computer systems today!) due to their connotations, we're ruling out (by definition) **every** word that describes the state of 
no choice or obedience.
Every word that describes state of being forced to follow orders from something above would sooner or later fall into this category. And descendant of black american slaves are not in any kind special here to deserve "sacred cow" treatment by allowing them to blacklist any words (yes, this word is also **not** related to the skin color: https://en.wikipedia.org/wiki/Blacklisting#Controversy).

Following this trend we should eliminate words like "command", "rules" (udev - there were dozens of bloody rulers in the history of mankind), "**kill**" (are we glorifying this, seriously?!), "abort" (anyone read this nonsense? https://inclusive*naming.org/word-lists/tier-1/), "mark" and "tag" (weren't the slaves branded with iron?), "atomic" (immediate association with "bomb") etc.

What next - are we going to enforce some different than black mourning colors in western world because there are people that call themselves "black"?


To sum this up:
- "slave" - **not** invented to abuse dark-skinned,
- "blacklist" - **not** invented for dark-skinned,
- "black" - **not** even the color of skin of most of the "black people" (https://www.grida.no/resources/7125).

So maybe instead of changing entire world, we just:
- stop using the term "black" to name the people with dark skin (which is usually not black) - let them invent their own affirmative word for their race,
- stop referring mentally to historical events when using contemporary language.

@slyon: please obscure the link to this abusive content (https://inclusive*naming.org) as it violates my feelings about Slavic (we are white) history of slavery by mendacious and racist statement (/word-lists/tier-1/_master-slave/: "*The terms master/slave are harmful to Black and people of color contributors and employees.*").

**Comment by g0tar** [comment]
@bluca - unfortunately I've started writing and posted before reading your threats, so apparently I'm in trouble now.
